mir_connector/inorbit_mir_connector/src/connector.py

```diff
@@ -113,92 +113,111 @@ def _inorbit_command_handler(self, command_name, args, options):
                 - `metadata` is reserved for the future and will contains additional
                 information about the received command request.
         """
-        if command_name == COMMAND_CUSTOM_COMMAND:
-            self._logger.info(f"Received '{command_name}'!. {args}")
-            script_name = args[0]
-            script_args = args[1]
-            # TODO (Elvio): Needs to be re designed.
-            # 1. script_name is not standarized at all
-            # 2. Consider implementing a callback for handling mission specific commands
-            # 3. Needs an interface for supporting mission related actions
-            if script_name == "queue_mission" and script_args[0] == "--mission_id":
-                self.mission_tracking.mir_mission_tracking_enabled = (
-                    self._robot_session.missions_module.executor.wait_until_idle(0)
-                )
-                self.mir_api.queue_mission(script_args[1])
-            elif script_name == "run_mission_now" and script_args[0] == "--mission_id":
-                self.mission_tracking.mir_mission_tracking_enabled = (
-                    self._robot_session.missions_module.executor.wait_until_idle(0)
-                )
-                self.mir_api.abort_all_missions()
-                self.mir_api.queue_mission(script_args[1])
-            elif script_name == "abort_missions":
-                self._robot_session.missions_module.executor.cancel_mission("*")
-                self.mir_api.abort_all_missions()
-            elif script_name == "set_state":
-                if script_args[0] == "--state_id":
-                    state_id = script_args[1]
-                    if not state_id.isdigit() or int(state_id) not in MIR_STATE.keys():
-                        self._logger.error(f"Invalid `state_id` ({state_id})")
-                        options["result_function"]("1")
-                        return
-                    state_id = int(state_id)
-                    self._logger.info(
-                        f"Setting robot state to state {state_id}: {MIR_STATE[state_id]}"
+        try:
```

Comment:
Hey who calls this function (`_inorbit_command_handler`) ? Can we move the try/except and handling of the error to the caller ? Seems this kind of error handling will be useful to every connector

Reply:
This is an overwrite of the connector method, which then registers it on the edge sdk. The callback could be wrapped in a try/except [here](https://github.com/inorbit-ai/inorbit-connector-python/blob/0c9c8fec39b9a4a05316e68067c4e0f6c5cb36d2/inorbit_connector/connector.py#L116-L122) in the Connector instead of the edge-sdk, to respect abstraction levels, achieving the exact same result:
```python
    def _register_custom_command_handler(self, handler: callable) -> None:
        """Register a custom command handler.

        Args:
            handler (Callable): The custom command handler.
        """

        def handler_wrapper(command_name: str, args: list, options: dict):
            try:
                handler(command_name, args, options)
            except Exception as e:
                self._logger.error(f"Error handling command {command_name}: {e}")
                self._logger.error(
                    f"Failed to execute command '{command_name}' with args {args}. "
                    f"Exception:\n{e}"
                )
                options["result_function"](
                    "1",
                    execution_status_details=(
                        "An error occured executing custom command"
                    ),
                    stderr=str(e),
                )

        self._robot_session.register_command_callback(handler_wrapper)
```

Do you think that would be a better solution?

Reply:
I really like that solution!

Reply:
@miguelgarcia https://github.com/inorbit-ai/inorbit-connector-python/pull/13

This PR still contains useful changes